### ISSUE TITLE
Images not showing in sample app

### ISSUE THREAD

**Issue by jamesnocentini** [couchbaselabs/react-native-couchbase-lite#102]
From @npomfret:

> ok, cool.  it's working on android, but without images as you mentioned.  Once issue PR 7791 get's added to RN they should start working.  In the mean time, I added a work around into the android API.  If you use null/null for the credentials it turns off authentication and also binds the listener to localhost only as a security precaution to prevent off-device access (so I think it works in a similar way to the iOS native listener).

**Comment by npomfret** [comment]
Looks like there's some work happening in RN core that might get this fixed https://github.com/facebook/react-native/pull/7791#issuecomment-270055174

**Comment by jamesnocentini** [comment]
@npomfret Excellent!

**Comment by npomfret** [comment]
https://github.com/facebook/react-native/pull/7791 has been merged

**Comment by npomfret** [comment]
Looks like we can get this working now: https://github.com/facebook/react-native/commit/8c0e6ecfc0e8b8ceb0471739ac2db53772d538cb

**Comment by jamesnocentini** [comment]
Re-opening because if it does work it means we can remove the workaround from this plugin and use the REST API as first intended.

**Comment by npomfret** [comment]
I'll try and upgrade my project today to see if it works

**Comment by npomfret** [comment]
@jamiltz 

I've tested RN0.43 and adding an `Authorization` header to an `<Image ` is working.  When I get a chance I'll submit a PR to update the sample app and we can clean up the android implementation a bit.

I don't know if it's implemented in iOS, but we don't exactly need it because the native URI doesn't require any auth.

**Comment by jamesnocentini** [comment]
Ok sounds great!

**Comment by jamesnocentini** [comment]
By cleaning up the implementation, do you mean removing those two blocks?

- https://github.com/couchbaselabs/react-native-couchbase-lite/blob/master/android/src/main/java/me/fraserxu/rncouchbaselite/ReactCBLite.java#L197-L344
- https://github.com/couchbaselabs/react-native-couchbase-lite/blob/master/ios/ReactCBLite.m#L97-L195

**Comment by npomfret** [comment]
No, I added some stuff to allow the user to specify null username and parameter which had the effect of switching of authorisation altogether.  Without any auth the images worked in react-native.  But that's not needed now.  

"No-auth" could be a feature to keep I suppose, but no one has asked for it, and it's not implemented for iOS.  Easier to just remove it I guess.

**Comment by jamesnocentini** [comment]
Oh ok I got confused with another ticket. I don't remember what's the limitation and why we can't save an attachment through the REST API.

**Comment by npomfret** [comment]
That's right.  I've found a couple of other things like that.
